src/jdk.compiler/share/classes/com/sun/tools/javac/util/GraphUtils.java

```diff
@@ -251,9 +251,9 @@ public void visitNode(N node, StringBuilder buf) {
         }
 
         protected String formatProperties(Properties p) {
-            return p.toString().replaceAll(",", " ")
-                .replaceAll("\\{", "[")
-                .replaceAll("\\}", "]");
+            return p.toString().replace(',', ' ')
```

Comment:
can't there be more than one `,` in the `toString` of `Properties` object?

Reply:
I know that javac only ever uses two properties at most... but this class is rather general, and the proposed change would introduce subtle bugs if a third property is ever added to inference nodes -- or any other node using in javac implementation

Reply:
(separately, I'm also curious of when such a change makes a real impact -- `toDot()` is mostly used when debugging... but I'm ok with replacing complex methods with simpler forms -- where possible)

Reply:
> But `replace` replaces all occurrences already.

Ah yes - that's where I got confused - thanks. (`replace` vs. `replaceAll` vs. `replaceFirst` mix a number of different concerns - such as one vs. many and string vs. char).